benchmark/python/benchmark_hail/run/utils.py

```diff
@@ -142,51 +173,73 @@ def initialize(args):
 
     # make JVM do something to ensure that it is fresh
     hl.utils.range_table(1)._force_count()
+    logging.getLogger('py4j').setLevel(logging.CRITICAL)
+    logging.getLogger('py4j.java_gateway').setLevel(logging.CRITICAL)
+
+
+def run_with_timeout(b, max_time):
+    with timeout_signal(max_time):
+        try:
+            return timeit.Timer(b.run).timeit(1), False
+        except Py4JError as e:
+            if _timeout_state:
+                return max_time, True
+            raise
+        except BenchmarkTimeoutError as e:
+            return max_time, True
 
 
-def _run(benchmark: Benchmark, config: RunConfig, context):
+def _run(b: Benchmark, config: RunConfig, context):
+    _ensure_initialized()
     if config.verbose:
-        logging.info(f'{context}Running {benchmark.name}...')
+        logging.info(f'{context}Running {b.name}...')
     times = []
 
+    timed_out = False
+    failed = False
     try:
-        burn_in_time = timeit.Timer(benchmark.run).timeit(1)
-        if config.verbose:
-            logging.info(f'    burn in: {burn_in_time:.2f}s')
+        burn_in_time, burn_in_timed_out = run_with_timeout(b, config.timeout)
+        if burn_in_timed_out:
+            if config.verbose:
+                logging.warning(f'burn in timed out after {burn_in_time:.2f}s')
+            timed_out = True
+            times.append(float(burn_in_time))
+        elif config.verbose:
+            logging.info(f'burn in: {burn_in_time:.2f}s')
     except Exception as e:  # pylint: disable=broad-except
         if config.verbose:
-            logging.error(f'    burn in: Caught exception: {e}')
-        config.handler({'name': benchmark.name,
-                        'failed': True})
-        return
+            logging.error(f'burn in: Caught exception: {e}')
+        failed = True
 
     for i in range(config.n_iter):
+        if timed_out or failed:
+            continue
```

Comment:
As soon as `timed_out` is true, you skip the rest of the loop, right? Why not just `break` after setting `timed_out`?

Reply:
I could fail / timeout during the burn-in, which is separate

Reply:
Ah, right. Python scoping is confusing. But still, `continue` could be `break`, no?

Reply:
ah, yes, definitely